### PR TITLE
feat: Set JDK language level to 11

### PR DESCRIPTION
**Pull request by mortenoh**
## Summary

This PR updates DHIS2 to only support JDK 11 and up. This also means we can now use JDK 11 features inside of DHIS2.

* [x] Remove `jdk8` and `jdk11` profiles
* [x] Update language level to JDK 11
* [x] Simple usage of `var` in `SchemaControllerTest`

[TECH-810](https://jira.dhis2.org/browse/TECH-810)

### CODE REVIEW

**Comment by varl** [comment]
Relates to: https://github.com/dhis2/dhis2-core/pull/9077

If this ends support for JRE/JDK 8 entirely (also when deploying Core) then we want to merge #9077 before merging this so we don't publish broken images that will be auto-updated for users and leave them in a broken state.

**Comment by teleivo** [comment]
@mortenoh we should then also change Jenkinsfiles which use the JDK8 profile
There might also be some other places referencing jdk8 things
https://github.com/dhis2/dhis2-core/search?q=jdk8

**Comment by mortenoh** [comment]
> @mortenoh we should then also change Jenkinsfiles which use the JDK8 profile There might also be some other places referencing jdk8 things https://github.com/dhis2/dhis2-core/search?q=jdk8

This is done now

**Comment by teleivo** [comment]
Now that I see your change in the Dockerfile these should be removed as well https://github.com/dhis2/dhis2-core/search?q=Pjdk11 😄

**Review by radnov — APPROVED** [review]
LGTM 👍

**Review by netroms — APPROVED** [review]

File: dhis-2/build-dev.sh [hunk 1]
```diff
@@ -51,8 +51,8 @@ cp -f "$DIR/dhis-web/dhis-web-portal/target/dhis.war" "$ARTIFACTS/dhis.war"
 print "Build succeeded, creating Docker image $IMAGE:$TAG..."
 
 cd $ARTIFACTS
-sha256sum ./dhis.war > ./sha256sum.txt
```
Comment (teleivo) [inline dhis-2/build-dev.sh:54]: these 2 removals are probably not intentional

Comment (mortenoh) [reply dhis-2/build-dev.sh:54]: It's just a small formatting fix

**Comment by teleivo** [comment]
I can still find at least one place that references the JDK11 profile https://github.com/dhis2/dhis2-core/blob/de3e492f51b710bd88d37a3c6e67b6bf6d2511d9/dhis-2/run-api.sh#L52

**Comment by teleivo** [comment]
The GitHub workflows also reference JDK11 https://github.com/dhis2/dhis2-core/blob/de3e492f51b710bd88d37a3c6e67b6bf6d2511d9/.github/workflows/run-tests.yml#L25 I think every one of them. A search for JDK11/8 will show you the spots 😄

**Comment by teleivo** [comment]
We probably need to change

https://github.com/dhis2/dhis2-core/blob/ce2b9089626ed23442f1884319557a98abd26bc0/dhis-2/pom.xml#L508-L509
to something like

```java
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-compiler-plugin</artifactId>
    <version>3.8.0</version>
    <configuration>
        <release>11</release>
    </configuration>
</plugin>
```

https://winterbe.com/posts/2018/08/29/migrate-maven-projects-to-java-11-jigsaw/

**Pull request merged**